calico_containers/pycalico/ipam.py

```diff
@@ -455,10 +459,18 @@ def _auto_assign(self, ip_version, num, handle_id,
                 _log.info("Ran out of affine blocks for %s in pool %s",
                           hostname, pool)
                 break
-            ips = self._auto_assign_block(block_id,
-                                          num_remaining,
-                                          handle_id,
-                                          attributes)
+            try:
+                ips = self._auto_assign_block(block_id,
```

Comment:
If this is the last block and you get a KeyError (because the client creating this block hasn't written it yet) - shouldn't we be retrying rather than potentially creating a new affine block below.